app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -41,7 +46,12 @@ def google_billing
     begin
       provider = request.env["omniauth.auth"].dig('provider')
       self.class.validate_scopes_from_params(params, provider)
+      self.class.validate_host_header_on_callback(request.headers)
     rescue SecurityError => e
+      logger.error e.message
+      context = ErrorTracker.format_extra_context(params)
+      ErrorTracker.report_exception(e, @user, context)
+      MetricsService.report_error(e, request, @user)
```

Comment:
And maybe even add a boolean (default true) `report_to_metrics` to the new method to make error handling even easier